knora/dsplib/utils/xml_upload.py

```diff
@@ -730,35 +889,82 @@ def xml_upload(input_file: str, server: str, user: str, password: str, imgdir: s
 
         resource_bitstream = None
         if resource.bitstream:
-            img = sipi.upload_bitstream(os.path.join(imgdir, resource.bitstream.value))
+            img = sipi_server.upload_bitstream(os.path.join(imgdir, resource.bitstream.value))
             internal_file_name_bitstream = img['uploadedFiles'][0]['internalFilename']
             resource_bitstream = resource.get_bitstream(internal_file_name_bitstream, permissions_lookup)
 
         permissions_tmp = permissions_lookup.get(resource.permissions)
 
         try:
             # create a resource instance (ResourceInstance) from the given resource in the XML (XMLResource)
-            instance: ResourceInstance = res_classes[resource.restype](con=con,
-                                                                       label=resource.label,
-                                                                       iri=resource_iri,
-                                                                       permissions=permissions_tmp,
-                                                                       bitstream=resource_bitstream,
-                                                                       values=resource.get_propvals(res_iri_lookup,
-                                                                                                    permissions_lookup)).create()
+            resclass_type = resclass_name_2_type[resource.restype]
+            properties = resource.get_propvals(res_iri_lookup, permissions_lookup)
+            resclass_instance: ResourceInstance = resclass_type(
+                con=con,
+                label=resource.label,
+                iri=resource_iri,
+                permissions=permissions_tmp,
+                bitstream=resource_bitstream,
+                values=properties
+            )
+            resclass_instance = resclass_instance.create()
         except BaseError as err:
-            print(
-                f"ERROR while trying to create resource '{resource.label}' ({resource.id}). The error message was: {err.message}")
+            print(f"ERROR while trying to create resource '{resource.label}' ({resource.id}). "
+                  f"The error message was: {err.message}")
             failed_uploads.append(resource.id)
             continue
-
         except Exception as exception:
-            print(
-                f"ERROR while trying to create resource '{resource.label}' ({resource.id}). The error message was: {exception}")
+            print(f"EXCEPTION while trying to create resource '{resource.label}' ({resource.id}). "
```

Comment:
I would leave the error message as it was. For the user an error is an error. Exception is only the technical term for an error.